integration_tests/tests/s3.rs

```diff
@@ -461,3 +469,49 @@ fn test_get_object_with_metadata(client: &TestClient, bucket: &str, filename: &s
     let head_metadata = result.metadata.as_ref().expect("No metadata available");
     assert_eq!(metadata, head_metadata);
 }
+
+fn test_generate_presigned_url(region: &Region, credentials: &AwsCredentials, bucket: &str, filename: &str) {
+    use reqwest::header::ContentLength;
+    use rusoto_s3::util::PreSignedRequest;
+    fn check(mut res: reqwest::Response){
+        assert!(&res.status().is_success());
+        let size = res.headers().get::<ContentLength>().map(|ct_len| **ct_len).unwrap_or(0);
+        assert!(size > 0);
+        let mut buf: Vec<u8> = vec![];
+        res.copy_to(&mut buf).unwrap();
+        assert!(buf.len() > 0);
+    }
+    {
+        let req = GetObjectRequest {
+            bucket: bucket.to_owned(),
+            key: filename.to_owned(),
+            ..Default::default()
+        };
+        let presigned_url = req.get_signed_url(region, credentials);
+        println!("get object presigned url: {:#?}", presigned_url);
+        let res = reqwest::get(&presigned_url).unwrap();
```

Comment:
How can I check the presgined url?

* https://semaphoreci.com/matthewkmayer/rusoto/branches/pull-request-1018/builds/7

```
{
  code: 336031996,
  library: "SSL routines",
  function: "SSL23_GET_SERVER_HELLO",
  reason: "unknown protocol",
  file: "s23_clnt.c",
  line: 795
}
```

```
get object presigned url: "https://localhost:9000/rusoto-test-bucket-1523526532/test_file_1523526532?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ANTN35UAENTS5UIAEATD%2F20180412%2Fus-east-1%20with%20endpoint%20http%3A%2F%2Flocalhost%3A9000%2Fs3%2Faws4_request&X-Amz-Date=20180412T094858Z&X-Amz-Expires=3600&X-Amz-Signature=ac2f328021b8af31ef5f1e12e6748bc665ad3c095a7498094a6b780e912f7733&X-Amz-SignedHeaders=content-type%3Bhost%3Bx-amz-content-sha256%3Bx-amz-date&x-amz-content-sha256=&x-amz-date=20180412T094858Z"
thread 'test_all_the_things' panicked at 'called `Result::unwrap()` on an `Err` value: Error { kind: Io(Custom { kind: Other, error: Ssl(ErrorStack([Error { code: 336031996, library: "SSL routines", function: "SSL23_GET_SERVER_HELLO", reason: "unknown protocol", file: "s23_clnt.c", line: 795 }])) }), url: Some("https://localhost:9000/rusoto-test-bucket-1523526532/test_file_1523526532?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ANTN35UAENTS5UIAEATD%2F20180412%2Fus-east-1%20with%20endpoint%20http%3A%2F%2Flocalhost%3A9000%2Fs3%2Faws4_request&X-Amz-Date=20180412T094858Z&X-Amz-Expires=3600&X-Amz-Signature=ac2f328021b8af31ef5f1e12e6748bc665ad3c095a7498094a6b780e912f7733&X-Amz-SignedHeaders=content-type%3Bhost%3Bx-amz-content-sha256%3Bx-amz-date&x-amz-content-sha256=&x-amz-date=20180412T094858Z") }', libcore/result.rs:945:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at libstd/sys_common/backtrace.rs:59
             at libstd/panicking.rs:380
   3: std::panicking::default_hook
             at libstd/panicking.rs:390
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:576
   5: std::panicking::begin_panic
             at libstd/panicking.rs:537
   6: std::panicking::begin_panic_fmt
             at libstd/panicking.rs:521
   7: rust_begin_unwind
             at libstd/panicking.rs:497
   8: core::panicking::panic_fmt
             at libcore/panicking.rs:71
   9: core::result::unwrap_failed
  10: <core::result::Result<T, E>>::unwrap
  11: s3::test_generate_presigned_url
  12: s3::test_all_the_things
  13: <F as alloc::boxed::FnBox<A>>::call_box
             at libtest/lib.rs:1462
             at /checkout/src/libcore/ops/function.rs:223
             at /checkout/src/liballoc/boxed.rs:788
  14: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:102
```

Reply:
I think the problem here is that the URL shouldn't include `https` when it's targeting localhost.

We have some logic here that does it for the normal request signing: https://github.com/rusoto/rusoto/blob/master/rusoto/core/src/signature.rs#L125

You should be able to use this method (`SignedRequest::scheme`) from within `SignedRequest::generate_presigned_url`.